### ISSUE TITLE
Problem installing virtual Devices with Simulator 

### ISSUE THREAD

**Issue by ToXIc69** [nutechsoftware/alarmdecoder-smartthings#22]
At Step 9 under installing SmartApp i get the following error. 

> physicalgraph.app.exception.UnknownDeviceTypeException: Device type 'AlarmDecoder action button indicator' in namespace 'alarmdecoder' not found. @line 1079 (doCall)

**Comment by f34rdotcom** [comment]
Hay. Sorry for the delay been deep in code lately. It looks like the directions were wrong and it did not specify to install a few of the devices needed including 'AlarmDecoder action button indicator'. I just updated 'dev' branch README to fix it. Can you confirm you have all 5 devices loaded

AlarmDecoder network appliance
AlarmDecoder virtual contact sensor
AlarmDecoder virtual smoke alarm
AlarmDecoder action button indicator
AlarmDecoder status indicator

Just an FYI I am about to make some more changes soon to these devices. We are finding that Amazon Echo is limited on what types of devices it can connect with. My plan is to try and morph the current devices and double up on some. Currently the "Action Button" is not seen because it is a button and an indicator. I will be splitting these out into a Button and an Indicator as separate devices for say "ARMED AWAY"

**Comment by ToXIc69** [comment]
I dont have the 5 devices loaded but will review the readme and see where the missing steps are and hopefully get back on track. 

also for the arming and disarming i used virtual switches. i would like native voice arming but would like an option to disable the voice disarm.

**Comment by ToXIc69** [comment]
so i've removed everything and started over and now when i try to do the simulator part i get the following error 

java.util.concurrent.TimeoutException: Execution time exceeded 20 app execution seconds: 96002012739276 @line -1 (doCall)

**Comment by f34rdotcom** [comment]
Hmm. No line number and doCall. This looks like some crazy internal error inside of the ST cloud. Just for a test try logging in using an ANON browser tab to clear your session and see if this persists in a new clean browser session.

**Comment by ToXIc69** [comment]
can you re-publish the older version so i can manually install it and get my alarm control back?

**Comment by f34rdotcom** [comment]
> can you re-publish the older version so i can manually install it and get my alarm control back?
Not likely going to know what point exactly you had checked out. I could maybe restore to a given point in our "debug" branch and you could switch to that. Please email ad2usb@support.nutech.com and I will arrange to do a remote desktop tech support call to help figure this out.

**Comment by ToXIc69** [comment]
i did almost a month ago and never got a reply here's my ticket #51011

**Comment by ToXIc69** [comment]
Hey Sean so we messed up. You gave me 6 zones but I have 7 is use. How can
I add the sevent?

**Comment by j3tang** [comment]
I seem to have the exact same issue with Toxic69. 
Before I found this discussion though, I noticed the 5 devices and only installed the 3 that the instructions indicated. It didn't work and so I installed all 5, but I get the same errors afterwards, nothing to simulate, and leaving the page means losing unsaved info. 

I've only just setup the system this week and just have the webapp working but SmartThings integration isn't going so well.
